### PR TITLE
[#6101] Fix GenQuery operators for DATA_RESC_HIER (4-3-stable)

### PR DESCRIPTION
**Pull request by alanking**
Addresses #6101 
Cherry-picked from #6751 

Running tests

### CODE REVIEW

File: server/api/src/rsGenQuery.cpp [hunk 1]
```diff
@@ -405,39 +424,38 @@ irods::error strip_resc_hier_name_from_query_inp( genQueryInp_t* _inp, int& _pos
...
-            if(!ret.ok()) {
-                irods::log(PASS(ret));
+    try {
+        // iterate over tmp and replace resource group with resource name
+        for (int i = 0; i < tmpV.len; ++i) {
+            const int inx = inxs[i];
+            const char* val = vals[i];
+            if (inx == COL_D_RESC_HIER) {
+                const auto new_cond = get_resc_id_cond_for_hier_cond(val);
+                addInxVal(&_inp->sqlCondInp, COL_D_RESC_ID, new_cond.empty() ? "='0'" : new_cond.c_str());
```
Comment (korydraughn) [inline server/api/src/rsGenQuery.cpp:447]: Can `new_cond.c_str()` result in a dangling pointer when `new_cond` is destructed?

File: server/api/src/rsGenQuery.cpp [hunk 2]
```diff
@@ -650,7 +668,7 @@ rsGenQuery( rsComm_t *rsComm, genQueryInp_t *genQueryInp,
         // the zone name must be skipped when getZoneHintForGenQuery() derives it.
         if (getValByKey(&genQueryInp->condInput, ZONE_KW)) {
             if (!is_zone_name_valid(zone_hint_str)) {
-                rodsLog(LOG_ERROR, "%s:%d :: Unknown zone name [%s].", __func__, __LINE__, zone_hint_str.data());
+                log_api::error("[{}:{}] - Unknown zone name [{}].", __func__, __LINE__, zone_hint_str.c_str());
```
Comment (korydraughn) [inline server/api/src/rsGenQuery.cpp:671]: You can remove the `.c_str()` since `zone_hint_str` is a `std::string`.

**Review by korydraughn — COMMENTED** [review]
Got that first use of `std::span`!

File: server/api/src/rsGenQuery.cpp (hunk 1 above)
Comment (alanking) [reply server/api/src/rsGenQuery.cpp:447]: `addInxVal` makes a copy of the passed in string for `condInput` via `strdup`, so I don't think that is a concern here:
https://github.com/irods/irods/blob/ec93f905d05776e67870ff80a92aba4227b926bd/lib/core/src/rcMisc.cpp#L866

File: server/api/src/rsGenQuery.cpp (hunk 2 above)
Comment (alanking) [reply server/api/src/rsGenQuery.cpp:671]: Indeed. Should I port this back to main as well?

Comment (korydraughn) [reply server/api/src/rsGenQuery.cpp:671]: I think so primarily for cherry-picking reasons. Keeping them in sync makes dealing with multiple branches easier.

Comment (alanking) [reply server/api/src/rsGenQuery.cpp:671]: These changes are already merged in main. This *is* the cherry-pick. That being said, I don't mind porting this change over to main.

Comment (korydraughn) [reply server/api/src/rsGenQuery.cpp:671]: Oh, if this is identical to what's in main, then leave it as is. No need to change anything.

**Comment by alanking** [comment]
Tests passed

**Review by korydraughn — APPROVED** [review]

**Review by korydraughn — APPROVED** [review]

**Review by trel — APPROVED** [review]

**Pull request merged**